examples/bri-3/src/bri/zeroKnowledgeProof/capabilities/createCCSMAnchorHash/createCCSMAnchorHashCommand.handler.ts

```diff
@@ -9,18 +10,25 @@ export class CreateCCSMAnchorHashCommandHandler
 {
   constructor(
     private readonly agent: CCSMAnchorHashAgent,
-    private readonly storageAgent: CCSMAnchorHashStorageAgent,
+    private readonly ccsmStorageAgent: CCSMAnchorHashStorageAgent,
+    private readonly localStorageAgent: CCSMAnchorHashLocalStorageAgent,
   ) {}
 
   async execute(command: CreateCCSMAnchorHashCommand) {
     this.agent.throwErrorIfCCSMAnchorHashInputInvalid(command.document);
 
-    const newCCSMAnchorHash = this.agent.createNewCCSMAnchorHash(
-      command.ownerAccount.id,
+    const newDocument = await this.localStorageAgent.createNewDocument(
       command.document,
     );
 
-    await this.storageAgent.storeAnchorHashOnCCSM(newCCSMAnchorHash);
+    const newCCSMAnchorHash = await this.agent.createNewCCSMAnchorHash(
+      command.ownerAccount.id,
+      newDocument,
```

Comment:
The ownerAccount id is also passed because it is meta data related to the hash (i.e. who has initiated this state change and hash) and is required for creating the anchor hash object.

Reply:
I was refering to line 26 where the newDocument parameter is passed. Instead of passing that, we should pass command.document since the call to create the new document should happen later in the execution of the command.